releasenotes/notes/update-gate_map-visualizations-6ea907a0502fdc1a.yaml

```diff
@@ -0,0 +1,15 @@
+---
+prelude: >
+    The following modules in `qiskit-terra/qiskit/visualization/gate_map.py` have been 
+    updated to leverage graphviz plotting for gate maps:
+    - `plot_gate_map` 
+    - `plot_coupling_map`
+    - `plot_circuit_layout`
+    - `plot_error_map` 
+    This allows the plotting of backends with large number of qubits without causing
+    issues due to migration to matplotlib. 
+
```

Comment:
One thing I think we need to document here as an `upgrade` note is the API change associated with the return type from the functions modified here. This is potentially a breaking change for any users of these functions because what was previously a `matplotlib.Figure` is now a `PIL.Image` object which behaves very differently. So we need to document this as something that any existing users of `plot_gate_map()` etc will have to change when upgrading to a new version.

Actually just writing that now, I'm wondering if we want to use matplotlib's `imshow` to import the image data into a matplotlib figure and returning that everywhere is the best path here. That would let us continue to return a `Figure` object for backwards compatibility everywhere. What do you think?

Reply:
Yes, I agree. Also, since the `plot_error_map()`'s return value is still a matplotlib Figure, this will create inconsistency. I am changing the return types of all the functions back to matplotlib Figure.